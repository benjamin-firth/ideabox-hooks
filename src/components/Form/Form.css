.enter-form {
  display: flex;
}

input {
  margin-right: 15px;
}