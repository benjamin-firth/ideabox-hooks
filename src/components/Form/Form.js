import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      date: '',
      time: '',
      number: 0
    }
  }

  handleChange = (event) => {
    this.setState({ [event.target.name]: event.target.value});
    console.log(this.state);
  }

  render() {
    return(
      <section className="enter-form">
        <input 
          name="name" 
          value={this.state.name} 
          onChange={(e) => this.handleChange(e)} placeHolder="Name" />
        <input 
          name="date" 
          value={this.state.date} 
          onChange={(e) => this.handleChange(e)} placeHolder="Date" />
        <input 
          name="time" 
          value={this.state.time} 
          onChange={(e) => this.handleChange(e)} placeHolder="Time" />
        <input 
          name="number" 
          value={this.state.number} 
          onChange={(e) => this.handleChange(e)} placeHolder="Number of Guests" />
        <button 
          type="button" 
          onClick={() => this.props.addNewReservation(this.state.name, this.state.date, this.state.time, this.state.number)}>Make Reservation</button>
      </section>
    )
  }
}

export default Form;