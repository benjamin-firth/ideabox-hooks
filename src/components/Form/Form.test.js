import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Form />)
  })

  describe('Form Snapshots', () => {
    
    it('Should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })
  })

  describe('State changes for form', () => {
    it('Should update state when function is fired', () => {
      const mockEvent = {
        target: {
          name: 'name',
          value: 'John Adams'
        }
      }

      const initialState = {
        name: '',
        date: '',
        time: '',
        number: 0
      }

      wrapper.setState(initialState);
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('name')).toEqual('John Adams');
    })
  })

  describe('Click events for form', () => {

    it('should fire addNewReservation when clicked', () => {
      const mockAddNewReservation = jest.fn();
      wrapper = shallow(<Form addNewReservation={mockAddNewReservation} />);

      wrapper.find('button').simulate('click');
      expect(mockAddNewReservation).toHaveBeenCalled();
    })

  })

})