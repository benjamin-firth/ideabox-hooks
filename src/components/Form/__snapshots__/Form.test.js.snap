// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Form Snapshots Should match the snapshot 1`] = `
<section
  className="enter-form"
>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date"
    value=""
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="Time"
    value=""
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="Number of Guests"
    value={0}
  />
  <button
    onClick={[Function]}
    type="button"
  >
    Make Reservation
  </button>
</section>
`;
